src/main/java/org/intellij/sdk/codesync/codeSyncSetup/CodeSyncSetupAction.java

```diff
@@ -161,17 +88,66 @@ public void actionPerformed(@NotNull AnActionEvent e) {
                         String.format("Could not disconnect the repo. Error: %s", error.getMessage())
                     );
                 }
-            } else {
+                return;
+            case NOT_SYNCED:
+            case UNKNOWN:
                 if (PricingAlerts.getPlanLimitReached()) {
                     // Show Pricing alert dialog.
                     PricingAlerts.setPlanLimitReached(project);
                 } else {
                     CodeSyncSetup.setupCodeSyncRepoAsync(project, repoPath, repoName, true, false);
                 }
-            }
-        } else {
-            NotificationManager.getInstance().notifyError(Notification.REPO_UNSYNC_FAILED, project);
-            CodeSyncLogger.error("Could not disconnect the repo. 0 module returned for the given project.");
+                break;
+            case DISCONNECTED:
+                try {
+                    CodeSyncSetup.reconnectRepo(project, repoPath, repoName);
```

Comment:
Which endpoint does it call now?

Reply:
We open webapp, same behavior as the deactivated account handling PR.

Reply:
Wait, confused, webapp is used for Reactivating Account. We discussed using the PATCH API to Reconnect Repo instaed of /v1/init

Reply:
Oh sorry, yes, we are using PATCH here.

Reply:
Got it. Was the PATCH implementation already there? I didn't see that in this PR, maybe overlooked

Reply:
Yes, it was already being used.